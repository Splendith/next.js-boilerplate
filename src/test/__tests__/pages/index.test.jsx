/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import ReduxWrapper from 'src/test/components/ReduxWrapper';
import Index from '~/pages/index';

describe('With Enzyme', () => {
  it('should shows "Hello world!"', () => {
    const app = shallow(<Index />);

    expect(
      app
        .find('Title')
        .children()
        .text(),
    ).toEqual('Hello world!');
  });
});

describe('With Snapshot Testing', () => {
  it('shallow snapshot', () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshot = shallowRenderer.render(<Index />);
    expect(snapshot).toMatchSnapshot();
  });
  it('full snapshot', () => {
    const component = renderer.create(
      <ReduxWrapper>
        <Index />
      </ReduxWrapper>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
