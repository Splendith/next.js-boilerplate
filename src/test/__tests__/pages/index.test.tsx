/* eslint-env jest */
import React from 'react';
import { render } from '@testing-library/react';
import preloadAll from 'jest-next-dynamic';

import ReduxWrapper from '../../components/ReduxWrapper';
import Index from '~pages/index';

beforeAll(async () => {
  await preloadAll();
});

describe('With Enzyme', () => {
  it('should shows "Hello world!"', () => {
    const app = render(
      <ReduxWrapper>
        <Index />
      </ReduxWrapper>,
    );

    expect(app.findByText('Title')).toEqual('Hello world!');
  });
});

// describe('With Snapshot Testing', () => {
//   it('shallow snapshot', () => {
//     const shallowRenderer = new ShallowRenderer();
//     const snapshot = shallowRenderer.render(<Index />);
//     expect(snapshot).toMatchSnapshot();
//   });
//   it('full snapshot', async () => {
//     let component;
//     await act(async () => {
//       component = renderer.create(
//         <ReduxWrapper>
//           <Index />
//         </ReduxWrapper>,
//       );
//     });
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
