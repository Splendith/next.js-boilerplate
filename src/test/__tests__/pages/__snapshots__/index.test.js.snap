// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing full snapshot 1`] = `
<div
  className="Wrapper-sc-1axzgde-0 eajKZs"
>
  <div
    className="Box-sc-1axzgde-1 eDkwhl"
  >
    <h1
      className="Title-sc-1g21ci3-0 hTAgCE"
    >
      Hello world!
    </h1>
    <div
      className="Image-jn44e3-0 hvdTEA"
    >
      <img
        alt="Smile"
        src="test-file-stub"
      />
    </div>
    <a
      href="/test/5?queryId=8"
      onClick={[Function]}
    >
      route=“/test/5?queryId=8“
    </a>
    <br />
    <a
      href="/test/7?queryId=9"
      onClick={[Function]}
    >
      route=“testurl“ params={{ id: 7, queryId: 9 }}
    </a>
  </div>
  <div
    className="Box-sc-68600l-0 flGqvX"
  >
    <h5>
      Redux global state
    </h5>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Add count
    </button>
     
    <button
      className="btn btn-outline-secondary"
      onClick={[Function]}
      type="button"
    >
      Clear count
    </button>
    <br />
    <br />
    <div>
      Counter: 
      0
    </div>
    <div>
      Loading...
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing shallow snapshot 1`] = `
<CounterPage>
  <SideEffect(Head)>
    <title>
      Hello world!
    </title>
  </SideEffect(Head)>
  <Title>
    Hello world!
  </Title>
  <Image>
    <img
      alt="Smile"
      src="test-file-stub"
    />
  </Image>
  <LinkRoutes
    route="/test/5?queryId=8"
  >
    <a>
      route=“/test/5?queryId=8“
    </a>
  </LinkRoutes>
  <br />
  <LinkRoutes
    params={
      Object {
        "id": 7,
        "queryId": 9,
      }
    }
    route="testurl"
  >
    <a>
      route=“testurl“ params={{ id: 7, queryId: 9 }}
    </a>
  </LinkRoutes>
</CounterPage>
`;
